pkg/volume/volume.go

```diff
@@ -151,24 +151,30 @@ func getVolumeClassNames(volumeClasses map[string]adminv1.ProjectVolumeClassInst
 	return typed.SortedKeys(storageClassName)
 }
 
-func CopyVolumeDefaults(volumeRequest v1.VolumeRequest, volumeBinding v1.VolumeBinding, volumeDefaults v1.VolumeDefault) v1.VolumeRequest {
+func ResolveVolumeRequest(volumeRequest v1.VolumeRequest, volumeBinding v1.VolumeBinding, resolvedVolume v1.VolumeResolvedOffering) v1.VolumeRequest {
```

Comment:
The previous iteration of this function had one responsibility: to put the request, binding and default together to get a merged volume request for the volume to process.

It seems that now this doesn't completely "resolve" the volume. It resolves the request without the defaults and the default information is added later, if needed. In other words, it kind of only does have the job now.

My opinion, which can be completely disregarded, is that passing the default information here and using it to complete "resolve" the volume would be better. Some care would need to be taken so that this function can be uses in the various places it is being used, but I think it would make it more complete.

Reply:
I reworked the function to fully resolve the volume based on the previously resolved values, all compute class information, bindings specified by the user, etc. Thankfully it didn't require too many modifications in the other places where this function is called.